test/test_circuitzoo_purification.jl

```diff
@@ -1,44 +1,25 @@
-using QuantumSavory
-using QuantumSavory.CircuitZoo
-using Test
-using QuantumSavory.CircuitZoo: EntanglementSwap, Purify2to1, Purify3to1, Purify3to1Node, Purify2to1Node, PurifyStringent, StringentHead, StringentBody, PurifyExpedient, PurifyStringentNode, PurifyExpedient
+@testitem "Circuit Zoo Purification - throws" tags=[:circuitzoo_purification] begin
+    using QuantumSavory.CircuitZoo
+    using QuantumSavory.CircuitZoo: EntanglementSwap, Purify2to1, Purify3to1, Purify3to1Node, Purify2to1Node, PurifyStringent, StringentHead, StringentBody, PurifyExpedient, PurifyStringentNode, PurifyExpedient
 
+    @test_throws ArgumentError Purify2to1(:lalala)
+    @test_throws ArgumentError Purify3to1(:lalala, :X)
+    @test_throws ArgumentError Purify2to1Node(:lalala)
+    @test_throws ArgumentError Purify3to1Node(:X, :lalala)
+    @test_throws ArgumentError StringentHead(:lalala)
+    @test_throws ArgumentError StringentBody(:lalala)
 
-const bell = StabilizerState("XX ZZ")
-# or equivalently `const bell = (Z₁⊗Z₁+Z₂⊗Z₂)/√2`,
-# however converting to stabilizer state for Clifford simulations
-# is not implemented (and can not be done efficiently).
-
-
-# QOptics repr
-const perfect_pair = (Z1⊗Z1 + Z2⊗Z2) / sqrt(2)
-const perfect_pair_dm = SProjector(perfect_pair)
-const mixed_dm = MixedState(perfect_pair_dm)
-noisy_pair_func(F) = F*perfect_pair_dm + (1-F)*mixed_dm # TODO make a depolarization helper
-
-
-# Qclifford repr
-const stab_perfect_pair = StabilizerState("XX ZZ")
-const stab_perfect_pair_dm = SProjector(stab_perfect_pair)
-const stab_mixed_dm = MixedState(stab_perfect_pair_dm)
-stab_noisy_pair_func(F) = F*stab_perfect_pair_dm + (1-F)*stab_mixed_dm
-
-@test_throws ArgumentError Purify2to1(:lalala)
-@test_throws ArgumentError Purify3to1(:lalala, :X)
-@test_throws ArgumentError Purify2to1Node(:lalala)
-@test_throws ArgumentError Purify3to1Node(:X, :lalala)
-@test_throws ArgumentError StringentHead(:lalala)
-@test_throws ArgumentError StringentBody(:lalala)
-
-r = Register(30)
-for i in 1:30
-    initialize!(r[i], X1)
+    r = Register(30)
+    for i in 1:30
+        initialize!(r[i], X1)
+    end
+    @test_throws ArgumentError PurifyExpedient()(r[1], r[2], r[3:2:21]...)
+    @test_throws ArgumentError PurifyStringent()(r[1], r[2], r[3:2:21]...)
 end
-@test_throws ArgumentError PurifyExpedient()(r[1], r[2], r[3:2:21]...)
-@test_throws ArgumentError PurifyStringent()(r[1], r[2], r[3:2:21]...)
-
 
-@testset "2to1" begin
+@testitem "Circuit Zoo Purification - 2to1" tags=[:circuitzoo_purification] begin
+    include("setup_circuitzoo_purification.jl")
```

Comment:
> @testitem blocks cannot be nested, and I don't know of any way to share set-up code between them, so in test/test_circuitzoo_purification.jl I just kept the @testset blocks, but all nested within one @testitem. If someone has a recommendation for how to do this more properly I will update this.

I have this working by refactoring the setup code to a separate file.  This seems to be the solution endorsed in [this thread](https://discourse.julialang.org/t/how-does-the-testing-pane-work/115301/8):
> Yes, that feature is just not finished. For now, the only real option to share code between different `@testitems` is to put it into a separate file, and then include that file from each @testitem.

Reply:
That makes sense! Thanks!